server/consumer.go

```diff
@@ -213,14 +213,13 @@ var (
 
 // Calculate accurate replicas for the consumer config with the parent stream config.
 func (consCfg ConsumerConfig) replicas(strCfg *StreamConfig) int {
-	if consCfg.Replicas == 0 {
+	if consCfg.Replicas == 0 || consCfg.Replicas > strCfg.Replicas {
 		if !isDurableConsumer(&consCfg) && strCfg.Retention == LimitsPolicy {
 			return 1
```

Comment:
We only want this, default to 1, for the old school ephemerals, meaning replicas should be 0.

If it is not zero, but larger then the stream now, we should assume they picked it and we should not return 1 here, but what the new stream replicas is..